x/participation/types/params.go

```diff
@@ -107,8 +132,41 @@ func validateParticipationTierList(v interface{}) error {
 		return fmt.Errorf("invalid parameter type: %T", v)
 	}
 
-	// TODO implement validation
-	_ = participationTierList
+	tiersIndexMap := make(map[uint64]struct{})
+	lastTierID := uint64(0)
+	lastRequiredAllocations := uint64(0)
+	lastTierBenefits := TierBenefits{MaxBidAmount: sdk.ZeroInt()}
+	for _, tier := range participationTierList {
+		// check IDs are unique
+		if _, ok = tiersIndexMap[tier.TierID]; ok {
+			return fmt.Errorf("duplicated tier ID: %v", tier.TierID)
+		}
+		tiersIndexMap[tier.TierID] = struct{}{}
+
+		// check IDs are sorted
+		if lastTierID > tier.TierID {
+			return fmt.Errorf("tiers must be sorted by ID")
+		}
+		lastTierID = tier.TierID
+
+		if tier.RequiredAllocations <= lastRequiredAllocations {
+			return errors.New("required allocations must be strictly increasing and greater than zero")
+		}
+		lastRequiredAllocations = tier.RequiredAllocations
+
+		if err := validateNextTierBenefits(tier.Benefits, lastTierBenefits); err != nil {
+			return err
+		}
+		lastTierBenefits = tier.Benefits
```

Comment:
In my opinion, this shouldn't be a requirement. We can manage the sorting in the frontend, using `TierID` to represent the benefits make the structure more flexible

Reply:
I am not sure about that. if we don't do this here we could end up with a lot of bloat tiers that would hurt performance of the dynamic computation of allocations. enforcing this mitigates the problem (since tiers are forced to be somewhat "meaningful")

Reply:
> that would hurt performance of the dynamic computation of allocations

I don't get this part. Computing the available allocation would be fully performed in my mind before checking the tier list. You computed the total available allocation and then you check if the value is higher than the necessary allocation for the tier you want to benefit from

Reply:
fair, I think I haven't fully thought it through when I replied. The tier is provided in `MsgParticipate` so what I said is plain wrong.
I still think not enforcing any constraint on the tier list is less elegant, but quite frankly I have no strong opinion. 
So I'm open to removing these checks.

@aljo242 @Pantani what's your opinion?

Reply:
I agree with @lubtd as long as the front end sorts these things.  My main reasoning is that you get a bit more flexibility to add "in between" tiers